infrastructure/kube/keep-test/monitoring/README.md

```diff
@@ -0,0 +1,162 @@
+
+```bash
+ACCOUNT=$(gcloud info --format='value(config.account)')
+kubectl create clusterrolebinding owner-cluster-admin-binding \
+    --clusterrole cluster-admin \
+    --user $ACCOUNT
+```
+
+*! This role can be created by Owner in GCP IAM*
+
+3. Create `clusterRole`
+
+  ```bash
+  kubectl create -f clusterrole.yaml
+  ```
+
+#### II. Create `ConfigMap` to put configuration externally
+
+1. Execute `create` command to create configmap
+    
+  ```bash
+  kubectl create -f configmap.yaml
+  ```
+
+Some words to that `ConfigMap`:
+
+All configurations are hold in `prometheus.yml`
+
+Alert rules for Alertmanager are configured in `promethus.rules`
+
+By externalizing Prometheus configs to a Kubernetes config map, there is no need to build Prometheus image whenever it needs additional configuration or some to be removed. Updating the config map and restarting the Prometheus pods to apply the new configuration is everything that needs to be done.
+
+The config map with all the Prometheus scrape config and alerting rules gets mounted to the Prometheus container in `/etc/prometheus` location as prometheus.yaml and prometheus.rules files.
+
+Beside that there is another ConfigMap located, isolatd from the one that belongs to Prometheus. It's created for Grafana.
+
+All datasources are stored in `prometheus.yaml` file. It holds information about prometheus server location so that grafana would have that extension already configured
+
+Dashboards config is stored in `dashboards.yaml`
+
+For keep-client monitoring purpose `client-dashboards.json` holds template config for that service
+
+One ConfigMap YAML file contains two section of configmaps configuration for two different services. 
+
+#### III. Create Prometheus PersistentVolumeClaim
+
+Create a PVC object
+
+  ```bash
+  kubectl create -f prometheus-pvc.yaml
+  ```
+
+#### IV. Create Prometheus Deployment
+
+In this configuration, config map is mounted as a file inside /etc/prometheus as explained in the previous section.
+
+  ```bash
+  kubectl create  -f prometheus-deployment.yaml
+  ```
+
+#### IV. Create Prometheus Service
+
+This object will use `NodePort` type for accessing the Prometheus service by using additional VPN service that is routed to kubernetes network. 
+
+  ```bash
+  kubectl create -f prometheus-service.yaml
+  ```
+
+Service should be available under
+
+http://prometheus.monitoring.svc.cluster.local:9090
+
+`prometheus` and `monitoring` in FQDN came from service object:
+
+```
+metadata:
+  name: prometheus
+  namespace: monitoring
+```
+
+
+### Grafana
+
+#### I. Create Grafana ConfigMap
+
+   ```bash
+   kubectl create -f grafana-configmap.yaml
+   ```
+
+Notice the **url** key in the ConfigMap file that refers to prometheus FQDN that comes from combination of `metadata` in the prometheus-service YAML file.
+
+#### II. Create Grafana PersistentVolumeClaim
+
+  ```bash
+  kubectl create -f grafana-pvc.yaml
```

Comment:
Same question as for Prometheus PersistentVolumeClaim: is the `storageclass.yaml` automatically resolved?